src/keys.ts

```diff
@@ -72,24 +75,46 @@ export function validateRootFingerprint(rootFingerprint) {
  * console.log(xpub.toBase58()) // returns base58 check encoded xpub
  */
 export class ExtendedPublicKey extends Struct {
-  constructor(options) {
+  path?: string;
+  sequence?: number[];
+  index?: number;
+  depth?: number;
+  chaincode?: string;
+  pubkey?: string;
+  parentFingerprint?: string;
+  network?: NETWORKS;
+  version?: KeyVersions;
+  rootFingerprint?: string;
+  base58String?: string;
+
+  constructor(options: Partial<ExtendedPublicKey>) {
     super();
     if (!options || !Object.keys(options).length) {
       return this;
     }
-
-    const pathError = validateBIP32Path(options.path);
-    assert(!pathError.length, pathError);
-    this.path = options.path;
-    this.sequence = bip32PathToSequence(this.path);
-    this.index = this.sequence[this.sequence.length - 1];
-    this.depth = this.path.split("/").length - 1;
+    if (options.path) {
+      const pathError = validateBIP32Path(options.path);
+      assert(!pathError.length, pathError);
+      this.path = options.path;
+      this.sequence = bip32PathToSequence(this.path);
+      this.index = this.sequence[this.sequence.length - 1];
+      this.depth = this.path.split("/").length - 1;
+    } else {
+      assert(
+        options.depth && options.index,
+        'Either an absolute bip32 path or index and depth are required to create ExtendedPublicKey'
+      )
+      this.depth = options.depth;
+      this.index = options.index;
```

Comment:
how bout a test to keep us at 100 coverage

Reply:
haha. good catch!